src/parser_urdf_TEST.cc

```diff
@@ -994,6 +995,1096 @@ TEST(URDFParser, ParseWhitespace)
+          </inertial>
+        </link>
+        <joint name='joint1_2' type='fixed'>
+          <parent link='link1' />
+          <child  link='link2' />
+          <origin xyz='0.0 0.0 0.0' rpy='0.0 0.0 1.57'/>
+        </joint>
+        <gazebo reference='joint1_2'>
+          <disableFixedJointLumping>true</disableFixedJointLumping>
+        </gazebo>
+        <gazebo reference='joint1_2'>
+          <preserveFixedJoint>true</preserveFixedJoint>
+        </gazebo>
+      </robot>)";
+
+    sdf::URDF2SDF parser;
+    sdf::ParserConfig defaultConfig;
+    tinyxml2::XMLDocument sdfResult;
+    parser.InitModelString(str, defaultConfig, &sdfResult);
+
+    // fixed joints to be reduced, but link1 is a root link with no fixed parent
+    // joint, conversion to frame still fails
+    EXPECT_PRED2(sdf::testing::contains, buffer.str(),
+        "urdf2sdf: link[link1] has no <inertial> block defined, but does not "
+        "have a fixed parent joint, unable to be converted into a frame in "
+        "sdf");
+    EXPECT_PRED2(sdf::testing::contains, buffer.str(),
+        "urdf2sdf: link[link1] has no <inertial> block defined, [1] children "
+        "links ignored");
+    EXPECT_PRED2(sdf::testing::contains, buffer.str(),
+        "urdf2sdf: link[link1] has no <inertial> block defined, [1] children "
+        "joints ignored");
+    EXPECT_PRED2(sdf::testing::contains, buffer.str(),
+        "urdf2sdf: link[link1] has no <inertial> block defined, not modeled in "
+        "sdf");
+
+    tinyxml2::XMLElement *sdf = sdfResult.FirstChildElement("sdf");
+    ASSERT_NE(nullptr, sdf);
+    tinyxml2::XMLElement *model = sdf->FirstChildElement("model");
+    ASSERT_NE(nullptr, model);
+    tinyxml2::XMLElement *frame = model->FirstChildElement("frame");
+    EXPECT_EQ(nullptr, frame);
+    tinyxml2::XMLElement *link = model->FirstChildElement("link");
+    EXPECT_EQ(nullptr, link);
+    tinyxml2::XMLElement *joint = model->FirstChildElement("joint");
+    EXPECT_EQ(nullptr, joint);
+  }
+
+  // root link child fixed joint with lumping turned off, don't convert into
+  // revolute joints, preserve fixed joints
+  {
+    // clear the contents of the buffer
+    buffer.str("");
+
+    std::string str = R"(
+      <robot name='test_robot'>
+        <link name='link1'/>
+        <link name='link2'>
+          <inertial>
+            <mass value='0.1' />
+            <origin rpy='1.570796326794895 0 0' xyz='0.123456789123456 0 0.0' />
+            <inertia ixx='0.01' ixy='0' ixz='0' iyy='0.01' iyz='0' izz='0.01' />
+          </inertial>
+        </link>
+        <joint name='joint1_2' type='fixed'>
+          <parent link='link1' />
+          <child  link='link2' />
+          <origin xyz='0.0 0.0 0.0' rpy='0.0 0.0 1.57'/>
+        </joint>
+        <gazebo reference='joint1_2'>
+          <disableFixedJointLumping>true</disableFixedJointLumping>
+        </gazebo>
+        <gazebo reference='joint1_2'>
+          <preserveFixedJoint>true</preserveFixedJoint>
```

Comment:
Since this `preserveFixedJoint` is set here, I don't see how this test block is different from the previous.